### PR TITLE
Telekinesis improvements/fixes

### PR DESCRIPTION
**Pull request by cacogen**
## About The Pull Request

1. Adds examine messages to the TK grab telling the player how to use the new TK controls (i.e. examining the TK grab no longer examines the focused item)
2. Fixes being unable to fire a gun with TK when wearing insulated gloves (also adds an arbitrary check to prevent hulks firing, because they use the same trait as insulated gloves to prevent them firing guns)
3. Makes it so guns with recoil or whose projectiles do damage are thrown back when firing, based on whichever value is largest (disablers for example are thrown back 3 tiles, based on 30 damage)
4. Improves code a bit for gun throwback when TK firing
5. Makes tk_firing() proc a bit more robust
6. Fixes point blank firing with TK being unable to cause knockbacks like regular firing

Not sure if this should update GBP or not, given these are my bugs but this is also ostensibly a balance change too.

## Why It's Good For The Game

1. Gives players a semi-intuitive way of learning how the new TK controls work for themselves. I don't think being able to examine the focused item through the grab was really used, so I don't think it'll be much of a loss to remove it.
2. This was something I missed while testing the pull that added TK firing. Doesn't make a lot of sense to prevent people from firing a gun when their fingers are too big if they're doing it with TK.
3. Adding more kickback helps balance TK firing a bit more, and just adds a bit of interesting variance and unpredictability to it
4. Early returns
5. Uses get() instead of just checking the gun's location is the user, which recursively checks every location the gun is in to try to find the mob firing it to verify it's not being fired with TK
6. Guns should presumably still knockback if you're firing with TK, this wasn't left out for balance reasons, just my own reluctance to duplicate a visible message

## Changelog

:cl:
qol: Examining the TK grab object now tells you the new controls for TK instead of examining the grabbed object
fix: TK firing is no longer blocked by fat fingers
balance: Guns with recoil or projectile damage are now thrown back slightly further when TK firing
code: Minor code improvements to TK firing
fix: Point blank firing with TK can now knockback like regular firing
/:cl:

### CODE REVIEW

**Comment by SuperNovaa41** [comment]
Turn TK into a component!!!
(I attempted this at one point but TK is like hard baked into the attack chain and would be very difficult to remove without side effects)

**Comment by Shadowflame909** [comment]
> Ah yes, now the only down-side of hulks (not being able to use ranged weapons) is fucking gone.
hulk got so nerfed its not something i'd worry about.

**Comment by Victor239** [comment]
> Examining the TK grab object now tells you the new controls for TK instead of examining the grabbed object

Make it a contextual screentip.

**Comment by private-tristan** [comment]
why do lasers have recoil in the first place?

**Comment by cacogen** [comment]
>Make it a contextual screentip.

There's no throw contextual screentip which is the least intuitive thing about how TK is now controlled.

I could add LMB and RMB screentips for the grab to indicate right-click is secondary attack though.

**Comment by cacogen** [comment]
@RuneQuester
>Ah yes, now the only down-side of hulks (not being able to use ranged weapons) is fucking gone.

Okay, I saw the agreement thumbs on that comment and made it so you can't fire with TK as a hulk.

File: code/modules/mob/living/carbon/human/human_helpers.dm [hunk 1]
```diff
@@ -99,13 +99,20 @@
...
 	. = ..()
+	if(HAS_TRAIT(src, TRAIT_NOGUNS))
+		to_chat(src, span_warning("You can't bring yourself to use a ranged weapon!"))
+		return FALSE
+
 	if(G.trigger_guard == TRIGGER_GUARD_NORMAL)
+		var/obj/item/tk_grab/telekinetic_grab = is_holding_item_of_type(/obj/item/tk_grab) //can fire with tk even if our fingers are fat
+		if(telekinetic_grab && telekinetic_grab.focus == G)
+			if(HAS_TRAIT(src, TRAIT_HULK))
+				G.balloon_alert(src, "not allowed!")
```
Comment (Fikou) [inline code/modules/mob/living/carbon/human/human_helpers.dm:110]:
```suggestion
				G.balloon_alert(src, "too angry to concentrate!")
```

**Pull request closed without merging**